tool/tsh/kube.go

```diff
@@ -892,13 +896,25 @@ func newKubeLoginCommand(parent *kingpin.CmdClause) *kubeLoginCommand {
 	}
 	c.Flag("cluster", clusterHelp).Short('c').StringVar(&c.siteName)
 	c.Arg("kube-cluster", "Name of the kubernetes cluster to login to. Check 'tsh kube ls' for a list of available clusters.").Required().StringVar(&c.kubeCluster)
+	c.Flag("as", "Configure custom Kubernetes user impersonation.").StringVar(&c.impersonateUser)
+	c.Flag("as-groups", "Configure custom Kubernetes group impersonation.").StringsVar(&c.impersonateGroups)
+	// TODO (tigrato): move this back to namespace once teleport drops the namespace variable.
+	c.Flag("kube-namespace", "Configure the default Kubernetes namespace.").Short('n').StringVar(&c.namespace)
+	c.Flag("all", "Generate a kubeconfig with every cluster the user has access to.").BoolVar(&c.all)
 	return c
 }
 
 func (c *kubeLoginCommand) run(cf *CLIConf) error {
 	// Set CLIConf.KubernetesCluster so that the kube cluster's context is automatically selected.
 	cf.KubernetesCluster = c.kubeCluster
 	cf.SiteName = c.siteName
+	cf.kubernetesImpersonationConfig = impersonationConfig{
+		kubernetesUser:   c.impersonateUser,
+		kubernetesGroups: c.impersonateGroups,
+	}
+	cf.kubeNamespace = c.kubeCluster
```

Comment:
@tigrato Was this intended to be assigned to `c.namespace`? Otherwise, `c.namespace` isn't referenced in this file after it's assigned via `StringVar`

Reply:
nice catch